File: problems/1496-lucky-numbers-in-a-matrix/lucky-numbers-in-a-matrix.js
/**
 * @param {number[][]} matrix
 * @return {number[]}
 */
const luckyNumbers  = (matrix) => {
    let minMax = new Set()
    for (let i = 0; i < matrix.length; i++) {
        let rowMin = Infinity;
        for (let j = 0; j < matrix[0].length; j++) {
            rowMin = Math.min(rowMin, matrix[i][j])
        }
        minMax.add(rowMin)
    }
    
    for (let j = 0; j < matrix[0].length; j++) {
        let colMax = - Infinity;
        for (let i = 0; i < matrix.length; i++) {
            colMax = Math.max(colMax, matrix[i][j])
        }
        if (minMax.has(colMax)) {
            return [colMax]
        } 
    }

    return []
};

File: problems/1496-lucky-numbers-in-a-matrix/lucky-numbers-in-a-matrix.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        minMax = set()
        for i in range(len(matrix)):
            minNum = inf
            for j in range(len(matrix[0])):
                minNum = min(minNum, matrix[i][j])
            minMax.add(minNum)
        
        for j in range(len(matrix[0])):
            maxNum = -inf
            for i in range(len(matrix)):
                maxNum = max(maxNum, matrix[i][j])
            if maxNum in minMax:
                return [maxNum]
        
        return []